src/agenteval/leaderboard/view.py

```
@@ -366,6 +367,79 @@ def construct_reproducibility_url(task_revisions: list[EvalRevision]) -> str | N
+        pretty_name = LB_MODEL_NAME_MAPPING[raw_name]
+        aliases.add(pretty_name)
+
+        # if the pretty name suggests it's unpinned
+        # include the pretty version without the date part
+        open_paren_index = pretty_name.rindex("(")
+        name_date = pretty_name[open_paren_index:].strip()
+        if name_date == "(unpinned)":
+            dateless_pretty_name = pretty_name[:open_paren_index].strip()
+            aliases.add(dateless_pretty_name)
+    return {a.lower() for a in aliases}
+
+
+def format_model_names_for_one_result(
+    raw_names: set[str], eval_spec: EvalSpec | None
+) -> dict[str, str]:
+    to_return: dict[str, str] = {}
+
+    if (
+        (eval_spec is not None)
+        and (eval_spec.model_args is not None)
+        and (isinstance(eval_spec.model_args, dict))
+        and ("reasoning_effort" in eval_spec.model_args)
+    ):
+        consider_eval_spec = True
+        spec_model_name_aliases = get_model_name_aliases(eval_spec.model)
+    else:
+        consider_eval_spec = False
+        spec_model_name_aliases = None
+
+    for raw_name in raw_names:
+        safe_name_option = LB_MODEL_NAME_MAPPING.get(raw_name, raw_name)
+        other_name_option = None
+
+        if consider_eval_spec:
+            # make mypy happy
+            assert eval_spec is not None
+            assert spec_model_name_aliases is not None
+            assert isinstance(eval_spec.model_args, dict)
+            raw_name_aliases = get_model_name_aliases(raw_name)
+            looks_like_same_model = (
+                len(raw_name_aliases.intersection(spec_model_name_aliases)) > 0
+            )
+            if looks_like_same_model:
+                reasoning_effort = eval_spec.model_args["reasoning_effort"]
+                other_name_option = adjust_model_name_for_reasoning_effort(
+                    model_name=safe_name_option,
+                    effort=reasoning_effort,
+                )
```

AFAIK this probably works for the specific experiments/agents we used, but in general for agents with multiple models, it's likely to run into mistakes (in particular, with an agent that tries cheap models first and escalates to higher reasoning if it detects a hard problem; this is a known strategy that we want to test at some point).

There are currently limitations for what we can do if usages are coming from outside Inspect, but for usages within Inspect the `"model"` events each have a property like:

```
"config": {
    "max_retries": 8,
    "max_connections": 8,
    "reasoning_effort": "medium"
},
```

which specifies the config for that particular model usage (as opposed to the global default config).  That example above is from the ReAct-o3 run on the leaderboard.  (There's also a `"call"` field with the exact request; could inspect directly that but maybe loses the benefit of any normalization Inspect tries to do).

I would like it if we can first check the direct model-usage config before falling back to the global default (and logging a corresponding warning).  Idk how hard that is to set up here but maybe it requires a lot of restructuring, in which case I think it's okay to merge this but we need to make a ticket to revisit when we start getting external submissions or testing mixed-reasoning agents.

We should also consider `reasoning_tokens` (Anthropic's finer-grained equivalent of OpenAI's `reasoning_effort`) at some point, but maybe that's off-topic for this ticket (though seems like it might be trivial to slot into the logic alongside reasoning_effort?).

---

thinking more...

So if we have `gpt-5` and `gpt-5 (unpinned)`, we will end up attaching reasoning effort to both of them?  But if you have those two different names, it basically means we really do have a mixture of different model configs (which could also have different reasoning settings).  I think if there's more than one variant of the same model used with `reasoning_effort`, this should be an error until we add per-usage effort extraction.  Presumably that wouldn't break anything with our current runs because we shouldn't have mixed variants of the reasoning models (not within the same run, anyway)?

---

oh one more thing: in theory this will be fine for my runs since I changed args per-model, but I believe it is technically possible to specify `--reasoning-effort` for models that don't actually do reasoning, e.g. gpt-4o, but would be silently ignore in the actual requests.  So we'd get a bogus name in that case.